common/types/encoding/codecv1/codecv1.go

```diff
@@ -204,7 +205,7 @@ func (c *DAChunk) Hash() (common.Hash, error) {
 }
 
 // NewDABatch creates a DABatch from the provided encoding.Batch.
-func NewDABatch(batch *encoding.Batch) (*DABatch, error) {
+func NewDABatch(batch *encoding.Batch, withCompression bool) (*DABatch, error) {
```

Comment:
Just to make sure I understand correctly: when `withCompression = true`, the only things that change are: `blob`, `BlobVersionedHash`, right? So `z`, `DataHash` etc. won't change.

Reply:
Yes.

Reply:
`z` would be changed because `z` is influenced by `blobVersionedHash`.